scripts/generate_drivers.py

```diff
@@ -40,15 +37,22 @@ class Register:
     name: str
     offset: int  # in bytes
     width: int  # in bits
-
-    @staticmethod
-    def make_register(name: str, offset: int, width: int) -> "Register":
+    fields: t.List[RegisterField]
+
+    @classmethod
+    def make_register(
+        cls,
+        name: str,
+        offset: int,
+        width: int,
+        fields: t.List[RegisterField],
+    ) -> "Register":
         if width not in (8, 16, 32, 64):
             raise Exception(f'Invalid register width {width}, for register '
                             f'{name}.\n'
                             f'Width should be not 8, 16, 32, or 64.\n'
                             f'Please fix the register width in DUH document.')
```

Comment:
I was thinking about it while looking over the code, but I decided that I wanted to hold off from doing that for now, since I've already touched a significant number of lines of code in this PR. I think in the common case, that assertion will still hold, and this is a script that's only meant to be run manually anyway, so we don't necessarily need to handle all cases.